### PR TITLE
Zachmu/as of

### PR DESCRIPTION
**Pull request by zachmu**
Dolt support for AS OF queries.

This won't build until https://github.com/liquidata-inc/go-mysql-server/pull/71 is merged and dependencies updated.

### CODE REVIEW

File: go/libraries/doltcore/sqle/database.go [hunk 1]
```diff
@@ -117,56 +125,114 @@ func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sq
...
+	root, err := db.rootAsOf(ctx, time)
 	if err != nil {
 		return nil, false, err
 	}
 
-	exactName, ok := sql.GetTableNameInsensitive(tblName, tableNames)
+	return db.getTable(ctx, root, tableName)
+}
 
+func (db *Database) rootAsOf(ctx *sql.Context, time interface{}) (*doltdb.RootValue, error) {
```
Comment (bheni) [inline go/libraries/doltcore/sqle/database.go]: don't love time as a variable name as it's the name of a standard go package.

File: go/libraries/doltcore/sqle/database.go [hunk 2]
```diff
@@ -117,56 +125,114 @@ func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sq
...
+	commit, ok := time.(string)
 	if !ok {
-		return nil, false, nil
+		panic("expected commit string")
+	}
+
+	cs, err := doltdb.NewCommitSpec(commit, "")
```
Comment (bheni) [inline go/libraries/doltcore/sqle/database.go]: if you want things like HEAD and HEAD^ to work then I believe you need the current branch.

File: go/libraries/doltcore/sqle/database.go (hunk 1 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/database.go]: Fixed.

File: go/libraries/doltcore/sqle/database.go (hunk 2 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/database.go]: Teach me senpai

File: go/libraries/doltcore/sql/sqltestutil/selectqueries.go [hunk 3]
```diff
@@ -766,6 +766,45 @@ var SelectDiffTests = []SelectTest{
...
+		ExpectedRows: []row.Row{
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(0), 1: types.String("Aaron"), 2: types.String("Son"), 3: types.String("123 Fake St"), 4: types.Uint(35)})),
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(1), 1: types.String("Brian"), 2: types.String("Hendriks"), 3: types.String("456 Bull Ln"), 4: types.Uint(38)},)),
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(2), 1: types.String("Tim"), 2: types.String("Sehn"), 3: types.String("789 Not Real Ct"), 4: types.Uint(37)})),
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(3), 1: types.String("Zach"), 2: types.String("Musgrave"), 3: types.String("-1 Imaginary Wy"), 4: types.Uint(37)})),
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(4), 1: types.String("Matt"), 2: types.String("Jesuele")})),
+			mustRow(row.New(types.Format_7_18, ReaddAgeAt4HistSch, row.TaggedValues{0: types.Int(5), 1: types.String("Daylon"), 2: types.String("Wilkins")})),
+		},
+		ExpectedSchema: ReaddAgeAt4HistSch,
+	},
```
[the remark this answers is not in the document]
Comment (zachmu) [reply go/libraries/doltcore/sql/sqltestutil/selectqueries.go]: Good catch, added tests here and in bats.

**Pull request merged**